wiring/src/spark_wiring_system.cpp

```diff
@@ -99,6 +100,66 @@ bool SystemClass::enableFeature(LoggingFeature) {
 }
 #endif
 
+#if HAL_PLATFORM_ASSETS
+
+spark::Vector<ApplicationAsset> SystemClass::assetsRequired() {
+    spark::Vector<ApplicationAsset> assets;
+    asset_manager_info info = {};
+    info.size = sizeof(info);
+    int r = asset_manager_get_info(&info, nullptr);
+    if (r) {
+        return assets;
+    }
+    SCOPE_GUARD({
+        asset_manager_free_info(&info, nullptr);
+    });
+    for (size_t i = 0; i < info.required_count; i++) {
+        assets.append(ApplicationAsset(&info.required[i]));
```

Comment:
@XuGuohui Sure, it might fail, the problem is that there is no easy way to deal with that in a sane manner here with the current API and even if we change it it doesn't bring any joy (e.g. if we return `SYSTEM_ERROR_NO_MEMORY`). What we do now is we simply not report the asset as available or required. If you under low memory conditions there is no way to read the assets anyway (memory required for decompression etc).